src/Agda2Hs/Compile/Term.hs

```diff
@@ -195,11 +195,9 @@ compileTerm v = do
       True  -> compileErasedApp es
       False -> (`app` es) . Hs.Con () =<< hsQName (conName h)
     Lit l -> compileLiteral l
-    Lam v b | usableModality v, getOrigin v == UserWritten -> do
-      unless (visible v) $ genericDocError =<< do
-        text "Implicit lambda not supported: " <+> prettyTCM (absName b)
+    Lam v b | keepArg v, getOrigin v == UserWritten -> do
       hsLambda (absName b) <$> underAbstr_ b compileTerm
-    Lam v b | usableModality v ->
+    Lam v b | keepArg v ->
       -- System-inserted lambda, no need to preserve the name.
       underAbstraction_ b $ \ body -> do
         unless (visible v) $ genericDocError =<< do
```

Comment:
Here, the error is kept (line 203), but it's actually dead code.

Reply:
Right, I'll remove it